src/tools/w3dviewer/datatreeview.cpp

```diff
@@ -0,0 +1,680 @@
+                        iterator->Current_Item_Name(),
+                        m_imageListIDs[ICON_ANIMATION],
+                        m_imageListIDs[ICON_ANIMATION],
+                        0,
+                        0,
+                        0,
+                        i,
+                        TVI_SORT);
+
+                    AssetInfoClass *info =
+                        new AssetInfoClass(iterator->Current_Item_Name(), ASSET_TYPE_ANIMATION, nullptr, nullptr);
+                    GetTreeCtrl().SetItem(newitem, TVIF_PARAM, nullptr, 0, 0, 0, 0, (LPARAM)info);
+                }
+            }
+        }
+    }
+}
+
+HTREEITEM CDataTreeView::FindItemByName(HTREEITEM item, const char *name)
+{
+    HTREEITEM item1 = 0;
+    HTREEITEM item2 = 0;
+    HTREEITEM item3 = GetTreeCtrl().GetChildItem(item);
+
+    if (item3 != 0) {
+        while (item1 == nullptr) {
+            if (lstrcmp(GetTreeCtrl().GetItemText(item3), name) == 0) {
+                item2 = item3;
+            }
+
+            item3 = GetTreeCtrl().GetNextSiblingItem(item3);
+
+            if (item3 == nullptr) {
+                return item2;
+            }
+
+            item1 = item2;
+        }
+    }
+
+    return item1;
+}
+
+HTREEITEM CDataTreeView::FindNextItem(HTREEITEM item, const char *name)
+{
+    HTREEITEM found = 0;
+
+    for (HTREEITEM i = GetTreeCtrl().GetNextSiblingItem(item); i != nullptr; i = GetTreeCtrl().GetNextSiblingItem(i)) {
+        if (found != nullptr) {
+            break;
+        }
+
+        AssetInfoClass *info = (AssetInfoClass *)GetTreeCtrl().GetItemData(i);
+
+        if (!m_restrictAnims || info != nullptr && !lstrcmpi(info->m_heirarchyName, name)) {
+            found = i;
+        }
+    }
+
+    return found;
+}
+
+HTREEITEM CDataTreeView::FindFirstItem(HTREEITEM item, const char *name)
+{
+    HTREEITEM found = 0;
+    for (HTREEITEM i = GetTreeCtrl().GetChildItem(item); i != nullptr; i = GetTreeCtrl().GetNextSiblingItem(i)) {
+        if (found != nullptr) {
+            break;
+        }
+
+        AssetInfoClass *info = (AssetInfoClass *)GetTreeCtrl().GetItemData(i);
+
+        if (!m_restrictAnims || info != nullptr && !lstrcmpi(info->m_heirarchyName, name)) {
+            found = i;
+        }
+    }
+
+    return found;
+}
+
+void CDataTreeView::SelectNext()
+{
+    HTREEITEM selected = GetTreeCtrl().GetSelectedItem();
+
+    if (selected != nullptr) {
+        HTREEITEM next = GetTreeCtrl().GetNextSiblingItem(selected);
+
+        if (next != nullptr) {
+            GetTreeCtrl().SelectItem(next);
+        }
+    }
+}
+
+void CDataTreeView::SelectPrev()
+{
+    HTREEITEM selected = GetTreeCtrl().GetSelectedItem();
+
+    if (selected != nullptr) {
+        HTREEITEM prev = GetTreeCtrl().GetPrevSiblingItem(selected);
+
+        if (prev != nullptr) {
+            GetTreeCtrl().SelectItem(prev);
+        }
+    }
+}
+
+void CDataTreeView::RestrictAnims(bool restrict)

```

Comment:
Restrict is a c++ keyword. Maybe can use different name to avoid wrong syntax highlighting in Editors.

Reply:
fixed.